### PR TITLE
[4.0] Normalize JS in All Categories views

### PR DESCRIPTION
**Pull request by SharkyKZ**
Fixes https://github.com/joomla/joomla-cms/issues/26695.

### Summary of Changes

Normalizes `categories-default.js` in `com_content`, com_contact` and `com_newsfeeds`, fixing several issues in All Categories views.

### Testing Instructions

Create categories with child categories for mentioned components.
Go to `List All Categories` views in frontend.
Click the `+` button to expand categories.
Check that plus icon changes to minus and that `aria-label` attribute of the button updates.

### Documentation Changes Required

No.

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 391d6a95a6c48a8e749b5f2d7665a81b4291d211<br /><br />great stuff - accessibility is great as well - thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27915">issues.joomla.org/tracker/joomla-cms/27915</a>.</sub>

**Comment by wilsonge** [comment]
@SniperSister @zero-24 can you look at RIPs here please

**Comment by infograf768** [comment]
@SharkyKZ 
If I do not mistake, these js contains the same code.
Would it be sensible or even possible to have only one file and load it for each component?

File: components/com_content/tmpl/categories/default.php [hunk 1]
```diff
@@ -19,36 +19,8 @@
 Text::script('JGLOBAL_EXPAND_CATEGORIES');
 Text::script('JGLOBAL_COLLAPSE_CATEGORIES');
 
-$js = <<<JS
-(function() {
-	document.addEventListener('DOMContentLoaded', function() {
-		var categories = [].slice.call(document.querySelectorAll('.categories-list'));
+HTMLHelper::_('script', 'com_content/categories-default.js', ['version' => 'auto', 'relative' => true]);
```
Comment (Fedik) [inline components/com_content/tmpl/categories/default.php]:
```php
$this->document->getWebAssetManager()
  ->registerAndUseScript('content.categories-default', 'com_content/categories-default.js', [], ['defer' => true], ['core']);
```

Comment (Fedik) [reply components/com_content/tmpl/categories/default.php]: well, but can ignore, I will replace it later

**Comment by SharkyKZ** [comment]
@infograf768 Yes, JS files are now identical. I just have no idea where to put the single reusable file.

File: components/com_content/tmpl/categories/default.php (hunk 1 above)
Comment (SharkyKZ) [reply components/com_content/tmpl/categories/default.php]: Is there some sort of naming convention to assets? E.g. in Debug plugin you used `plg.system.debug` so here should be `com.content.categories-default`?

Comment (Fedik) [reply components/com_content/tmpl/categories/default.php]: > Is there some sort of naming convention

nope, for now 🙈

but I thought about 
components: `name.blabla`
plugin, modules: `plg.name.blabla`, `mod.name.blabla`

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 391d6a95a6c48a8e749b5f2d7665a81b4291d211<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27915">issues.joomla.org/tracker/joomla-cms/27915</a>.</sub>

**Comment by jwaisner** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27915">issues.joomla.org/tracker/joomla-cms/27915</a>.</sub>

**Comment by Fedik** [comment]
> Yes, JS files are now identical. I just have no idea where to put the single reusable file.

place it under media/com_categories, name it somehow useful categories-shared-random-stuff.js and use it in other components :wink:

**Comment by laoneo** [comment]
Do we need here another round of tests or can RTC stay?

**Comment by SharkyKZ** [comment]
Set back to Pending, I guess. The tracker won't let me remove RTC label..

**Comment by astridx** [comment]
Do you like to have tests here? If so, could you please resolve the conflicts?

**Comment by SharkyKZ** [comment]
Conflicts solved.

**Comment by astridx** [comment]
I have tested this Patch.

1. git fetch origin pull/27915/head:j4/js/categories-default
2. git checkout j4/js/categories-default
3. No new install **but** I run npm ci.
4. I created categories with child categories for com_content, com_contact and com_newsfeed.
5. I created menu items that shows emty categories.
6. I could open and close the all views in the front end. 

newsfeed



I tested with the current ‎4.0.0-beta1-dev on Ubuntu with PHP 7.2 and MySQL 5.7.

**Comment by astridx** [comment]
I have tested this item :white_check_mark: successfully on ee48f1f789ab23b2e5d6bdd6576954f9591128fe<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27915">issues.joomla.org/tracker/joomla-cms/27915</a>.</sub>

**Comment by wilsonge** [comment]
Thanks! Nice work :)

**Pull request merged**